Source/Core/Core/PowerPC/BreakPoints.h

```diff
@@ -132,3 +134,39 @@ class MemChecks
   Core::System& m_system;
   bool m_mem_breakpoints_set = false;
 };
+
+class DelayedMemCheckUpdate final
+{
+public:
+  DelayedMemCheckUpdate(MemChecks* memchecks, bool update_needed = false)
+      : m_memchecks(memchecks), m_update_needed(update_needed)
+  {
+  }
+
+  DelayedMemCheckUpdate(const DelayedMemCheckUpdate&) = delete;
+  DelayedMemCheckUpdate(DelayedMemCheckUpdate&& other) = delete;
+  DelayedMemCheckUpdate& operator=(const DelayedMemCheckUpdate&) = delete;
+  DelayedMemCheckUpdate& operator=(DelayedMemCheckUpdate&& other) = delete;
+
+  ~DelayedMemCheckUpdate()
+  {
+    if (m_update_needed)
+      m_memchecks->Update();
+  }
+
+  DelayedMemCheckUpdate& operator|=(DelayedMemCheckUpdate&& other)
+  {
+    if (m_memchecks == other.m_memchecks)
```

Comment:
Should something happen if the pointers differ? An immediate `Update`? An `assert`?

Reply:
I think that's supposed to be `m_memchecks != other.m_memchecks`.

Reply:
I intentionally wrote it this way. If the two `m_memchecks` point to different objects, `other` will be destroyed at the end of the function and its destructor will call `Update` if needed. (I was considering going with an assert instead, though.)

Reply:
Oh, yeah, it makes sense.